check_license_source_files.sh

```diff
@@ -188,10 +188,9 @@ EOF
         cat "$license" >&2
         TEST_RESULT=1
     else
-        copyright_modified=$(echo "${CM} Copyright <copyright_year> Northern.tech AS" | sed "s/<copyright_year>/$modified_year/g")
-        copyright_file="$(head -n 1 "$file")"
-        if [ "$copyright_modified" != "$copyright_file" ]; then
-            echo >&2 "!!! FAILED license check on ${orig_file}; make sure copyright year matches last modified year of the file ($modified_year)"
+        copyright_year=$(grep -oP "${CM} Copyright \K\d{4} Northern.tech AS")
+        if [ $copyright_year -lt $added_year ]; then
```

Comment:
Since the failure condition is based on a positive match, any failure in parsing will lead to numeric error, and for this to pass in error. This includes when the license is simply missing. One possible fix is:

```bash
if [ $copyright_year -ge $added_year ]; then
    # Success!
    :
else
    echo >&2 "!!! FAILED license check on ${orig_file}; make sure copyright year is at least the year the file was first added to git ($added_year)"
    TEST_RESULT=1
fi
```